File: src/test/java/TestBase.java
import myLang.block.BlockContext;
import myLang.interpreter.MyLangInterpreter;
import myLang.response.MyLangException;
import myLangParser.MyLangLexer;
import myLangParser.MyLangParser;
import org.antlr.v4.runtime.CharStreams;
import org.antlr.v4.runtime.CommonTokenStream;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

public class TestBase {
    final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
    private final PrintStream originalOut = System.out;
    static {
        System.setProperty("line.separator","\r\n");
    }

    void run(String input){
        MyLangLexer lexer = new MyLangLexer(CharStreams.fromString(input));

        CommonTokenStream tokens = new CommonTokenStream(lexer);
        MyLangParser parser = new MyLangParser(tokens);
        try {
            new MyLangInterpreter().visitCompilationUnit(parser.compilationUnit());
        } catch (MyLangException e) {
            e.printStackTrace();
            Assert.fail();
        }
    }

    void runWithException(String input, String errorMsg){
        MyLangLexer lexer = new MyLangLexer(CharStreams.fromString(input));

        CommonTokenStream tokens = new CommonTokenStream(lexer);
        MyLangParser parser = new MyLangParser(tokens);
        try {
            new MyLangInterpreter().visitCompilationUnit(parser.compilationUnit());
            Assert.fail();
        } catch (MyLangException e) {
            //e.printStackTrace();
            Assert.assertEquals(errorMsg, e.getMessage());
        }
    }

    @Before
    public void setUp() {
        System.setOut(new PrintStream(outContent));
    }

    @After
    public void tearDown() {
        /*Class<?> blockContextClass = getClass().getClassLoader().loadClass(BlockContext.class.getCanonicalName());

        Object foo = blockContextClass.newInstance();*/
        BlockContext.clear();
        System.setOut(originalOut);
    }
}

File: src/test/kotlin/TestBase.kt
import myLangKotlin.block.BlockContext
import myLangKotlin.interpreter.MyLangInterpreter
import myLangKotlin.response.MyLangException
import myLangParser.MyLangLexer
import myLangParser.MyLangParser
import org.antlr.v4.runtime.CharStreams
import org.antlr.v4.runtime.CommonTokenStream
import org.junit.After
import org.junit.Assert
import org.junit.Before

import java.io.ByteArrayOutputStream
import java.io.PrintStream

open class TestBase {
    internal val outContent = ByteArrayOutputStream()
    private val originalOut = System.out

    internal fun run(input: String) {
        val lexer = MyLangLexer(CharStreams.fromString(input))

        val tokens = CommonTokenStream(lexer)
        val parser = MyLangParser(tokens)
        try {
            MyLangInterpreter().visitCompilationUnit(parser.compilationUnit())
        } catch (e: MyLangException) {
            e.printStackTrace()
            Assert.fail()
        }

    }

    internal fun runWithException(input: String, errorMsg: String) {
        val lexer = MyLangLexer(CharStreams.fromString(input))

        val tokens = CommonTokenStream(lexer)
        val parser = MyLangParser(tokens)
        try {
            MyLangInterpreter().visitCompilationUnit(parser.compilationUnit())
            Assert.fail()
        } catch (e: MyLangException) {
            //e.printStackTrace();
            Assert.assertEquals(errorMsg, e.message)
        }

    }

    @Before
    fun setUp() {
        System.setOut(PrintStream(outContent))
    }

    @After
    fun tearDown() {
        BlockContext.clear()
        System.setOut(originalOut)
    }

    companion object {
        init {
            System.setProperty("line.separator", "\r\n")
        }
    }
}
